src/concepts/pipeline/schemas.md

```diff
@@ -0,0 +1,207 @@
+# Generated Schemas
+
+## Overview
+
+Schemas describe the structure of ingested data. They are used in the pipeline to validate the types
+and values of data, or to define a table schema in a data store. We use a repository of JSON Schemas
+to sort incoming data into [`decoded` and `error`
+datasets](../../cookbooks/bigquery.md#projects-with-bigquery-datasets). We also generate BigQuery
+table schemas nightly from the JSON Schemas.
+
+This section is intended for those who want modify the process of generating and applying schemas
+in various components of the data pipeline.
+
+```mermaid
+graph TD
+
+%% Nodes
+subgraph GCP
+  bigquery
+end
+
+subgraph mozilla-pipeline-schemas
+  master
+  schemas(generated-schemas)
+end
+
+generator(mozilla-schema-generator)
+transpiler(jsonschema-transpiler)
+probe-info(probe-scraper)
+ingestion(gcp-ingestion)
+airflow(telemetry-airflow)
+
+
+%% Node hyperlinks
+click bigquery "../../cookbooks/bigquery.html"
+click master "https://github.com/mozilla-services/mozilla-pipeline-schemas"
+click schemas "https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/generated-schemas"
+click generator "https://github.com/mozilla/mozilla-schema-generator"
+click transpiler "https://github.com/mozilla/jsonschema-transpiler"
+click probe-info "https://github.com/mozilla/probe-scraper"
+click ingestion "https://mozilla.github.io/gcp-ingestion/ingestion-beam/"
+click airflow "https://github.com/mozilla/telemetry-airflow"
+
+
+%% Edges
+master --> |git clone| generator
+transpiler --> |used by| generator
+probe-info --> |used by| generator
+generator --> |scheduled by| airflow
+airflow --> |run nightly| schemas
+
+schemas --> |defines table| bigquery
+schemas --> |defines is valid| ingestion
+schemas --> |defines normalization| ingestion
+ingestion --> |inserts into| bigquery
+```
+
+**Figure**: _An overview of generated schemas. Click on a node to navigate to the relevant
+repository or documentation._
+
+## Schema Repository
+
+```mermaid
+graph LR
+
+subgraph mozilla-pipeline-schemas
+  subgraph origin/master
+    templates -->|cmake| schemas
+  end
+
+  subgraph origin/generated-schemas
+    schemas -->|mozilla-schema-generator| artifact(schemas)
+  end
+end
+```
+
+**Figure**: _Template schemas are built locally to generate static JSON Schema. On a regular basis,
+the Mozilla Schema Generator is run to generate BigQuery schemas._
+
+Refer to [Sending a Custom Ping](../../cookbooks/new_pings.md) for an in-depth guide for adding new
+schemas to the repository.
+
+### Mozilla Schema Generator
```

Comment:
A few key points here:
- JSON schemas are generated and used by the transpiler for BQ schema generation, but _not_ for validation
- Schema aliases are generated here
- A few fields in the environment are altered as well